### PR TITLE
[FIX] web_editor: fix non-deterministic tests

### PR DESCRIPTION
**Pull request by dmo-odoo**
Those tests were skipped because they were failing
non-deterministically.

This commit aims to make them deterministic by putting back some
guard promises that were lost during the jquery update adaptation.

This commit also fixes a lingering issue with one of the asserts.
We observed that sometimes, the "alt" and "title" attributes where
both present on the element but displayed in different order in
the string returned by wysiwyg.getValue(). This is because there
is no set order in the attribute returned by jQuery.html().
Because of that, I replaced the simple sring test by a thorough
manual comparison of the result that is expected to be equivalent.

### CODE REVIEW

File: addons/web_editor/static/tests/wysiwyg_tests.js [hunk 1]
```diff
@@ -3011,6 +3018,44 @@ QUnit.module('Media', {
                 return this._super(route, args);
             },
         };
+
+        testUtils.mock.patch(AltDialog, {
+            init: function () {
+                var self = this;
+                this._super.apply(this, arguments);
+                defAltDialogInit = new Promise (function (resolve) {
```
Comment (aab-odoo) [inline addons/web_editor/static/tests/wysiwyg_tests.js]:
````
defAltDialogInit = new Promise (function (resolve) {
    self.opened(resolve);
});
````
or even
````
defAltDialogInit = this._opened;
````

**Comment by dmo-odoo** [comment]
@aab-odoo @VincentSchippefilt y'a encore un assert qui foire de façon non-déterministe, mais il faut vraiment beaucoup de retry pour le faire apparaitre donc je ne l'aperçois que maintenant... pas sûr que ce soit dû à la branche jquery par contre

**Comment by aab-odoo** [comment]
@dmo-odoo you could ask moc to search for the failing assert into failed builds, he would be able to tell if it's only in our branch or not

**Comment by dmo-odoo** [comment]
@aab-odoo @VincentSchippefilt I forgot to unpatch so that might be the cause of the error I observed lately. I will apply aab's suggestion and rerun these two tests 50 more times... :(

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by dmo-odoo** [comment]
On dirait bien qu'on a un autre test non déterministe... ^^'

**Comment by aab-odoo** [comment]
robodoo retry

**Pull request closed without merging**